### PR TITLE
Game logic for the tic tac toe game

### PR DESCRIPTION
**Pull request by collinsugwu**
**What this PR does**

1. welcome the players and instruct them
2. assign symbols to players
3. display boards for players to see
4. validate players' input and switch turn amongst players
5. check for a winner or a draw
6. display winner

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -2,4 +2,94 @@
...
+    else
+      puts 'Position taken, choose another position'
+    end
+  end
+end
+
+interface = Interface.new
+interface.welcome
+interface.play
+interface.display_winner
```
Comment (maelfosso) [inline bin/main.rb]: That file, **bin/main.rb** must be executable
We should run the game with only **bin/main.rb**

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  def initialize
+    @position = Array.new(9, ' ')
+    @turn = 1
+  end
+
+  def position=(position)
+    @position[(position.to_i - 1)] = @turn.odd? ? 'X' : 'O'
+    @turn += 1
+  end
```
Comment (maelfosso) [inline lib/board.rb]: Nice :+1:

Comment (maelfosso) [reply lib/board.rb]: Use symbol **:X / :O** instead of `'X'` or `'O'`

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  # returns 1 if player 1 wins, returns 2 if player 2 wins, returns 0 if there is no winner, and returns 3 if there is a draw
+  def winner
+    3.times { |i| return 1 if @position[0 + (i * 3)] == 'X' && @position[1 + (i * 3)] == 'X' && @position[2 + (i * 3)] == 'X' }
+    3.times { |i| return 1 if @position[0 + i] == 'X' && @position[3 + i] == 'X' && @position[6 + i] == 'X' }
+    2.times { |i| return 1 if @position[0 + (i * 2)] == 'X' && @position[4] == 'X' && @position[8 - (i * 2)] == 'X' }
+    3.times { |i| return 2 if @position[0 + (i * 3)] == 'O' && @position[1 + (i * 3)] == 'O' && @position[2 + (i * 3)] == 'O' }
+    3.times { |i| return 2 if @position[0 + i] == 'O' && @position[3 + i] == 'O' && @position[6 + i] == 'O' }
+    2.times { |i| return 2 if @position[0 + (i * 2)] == 'O' && @position[4] == 'O' && @position[8 - (i * 2)] == 'O' }
```
Comment (maelfosso) [inline lib/board.rb]: Even if it's working, coding like this it's not good.
I suggest you do something like this
```ruby
3.times { |i| return 1 if all_equal(@position[0 + (i*3)..2 + (i*3)], :X) }
.... 
```
with **all_equal** define like
```ruby
def all_equal(subarray, player_symbol)
    return true if subarray.uniq.size == 1 && subarray.uniq == player_symbol
  end
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Small changes required

Hello @collinsugwu @jairjy 

Really nice code :smile: The Game is OK :clap: 
However the code need small changes. Please, read the comments left below to know more about them.

Kindly make the changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by stratospherique — APPROVED** [review]
### Nice fixes that you have implemented there :clap: . Congrats, you have passed the third milestone :heavy_check_mark:  and you may merge and move on to the next :+1: :+1: 

Keep up the good work :muscle: 
**Ahmed Mahfoudh**

**Pull request merged**